projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,591 @@
+	- In case of success, display a success message.
+	- In case of failure, display an error feedback.
+
+- **Error Management**:
+	- An internal error can happen during the wallet loading process. In this situation a user has no option other than try again later.
+    - [Suggestion] After 3 failing attempts the error feedback can ask the user to send the
+      error to the team.
+
+## Reading Nano Contract Transactions
+
+```mermaid
+graph LR
+    A[Main] -->|This Flow| B[Nano Contract Transactions]
+    A -->C[Nano Contract Registration]
+```
+
+### Screen: Main
+[screen-main]: #screen-main
+
+After a user ops-in the *Nano Contract* feature and the wallet loading completes, the Main screen displays the **Token List** component. Additionally, a user can choose to view the **Nano Contract List** component through a clearly marked option.
+
+Each list component offers a registration option:
+- "Register Token"
+- "Register Nano Contract"
+
+Selecting "Register Nano Contract" option navigates the user to the **Nano Contract Registration** screen.
+
+##### Design Bill
+- **Placement**:
+	- Position the option to display the *Nano Contract List* in a prominent area of the Main screen for easy accessibility.
+	- Place the action buttons "Register Token" and "Register Nano Contract" at the same context of their respective lists for clear visibility and access.
+  
+- **Functionality**:
+	- The *Nano Contract List* display option should reveal all registered nano contracts in the wallet when selected.
+	- The "Register Token" button navigates the user to the Token Registration screen.
+	- The "Register Nano Contract" button navigates the user to the **Nano Contract Registration** screen for adding a new Nano Contract.
+  
+- **Visibility Conditions**:
+	- The *Nano Contract List* display option should only be visible and active for users who have opted into the nano-contract feature.
+	- The "Register Nano Contract" button should be visible only when the *Nano Contract List* is being displayed.
+
+- **Error Management**:
+	- An internal error can happen during the wallet loading process. In this
+      situation a user has no other option than try again later.
+    - [:pushpin: Suggestion] After 3 failing attempts the error feedback can ask the user to send the
+      error to the team.
+
+### Component: Token List
+[component-token-list]: #component-token-list
+
+This component displays a list of registered *Tokens* on the *Dashboard* screen, allowing users to select and interact with each token. By clicking on a token item the user is navigated to the *Token Transactions* component.
+
+>[!NOTE]
+>The actual implementation name for the "Token List" component is `TokenSelect`, which is found in the *Dashboard Screen*. It is more convenient to call "Token List" in this design because the component presents a list of tokens, which can be selected, but the selection is just a possible action, and doesn't convey the shape of component when rendered.
```

Comment:
I see. Thanks for the context! I believe we can improve the semantics of components and do some refactoring. Do you think this worths a project, or it can be done as a KTLO?